.github/workflows/pr-title-lint.yml

```diff
@@ -0,0 +1,87 @@
+name: Lint Pull Request Title
+
+on:
+  pull_request:
+    types:
+      - opened
+      - edited
+      - reopened
+      - synchronize
+
+permissions:
+  contents: read
+
+jobs:
+  lint_commit:
+    runs-on: ubuntu-latest
+    permissions:
+      contents: read
+      pull-requests: write
+
+    steps:
+      - name: Checkout repository
+        uses: actions/checkout@v4
+
+      - name: Install Commitlint
+        shell: bash
+        run: npm install --location=global @commitlint/cli@17.1.1
+
+      - name: Lint Commit
+        shell: bash
+        env:
+          GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
+        run: |
+          set -ex
+          PR_TITLE="${{ github.event.pull_request.title }}"
+
+          ERR='
+            The PR title `${{ github.event.pull_request.title }}` does not follow the Calimero-Network Commit Message Convention
+            https://github.com/calimero-network/core/blob/master/CONTRIBUTING.md#commit-message-style
+
+            Our PR title should have the following structure.
+
+              <type>(<scope>): <subject>
+              <BLANK LINE>
+              <body>
+              <BLANK LINE>
+              <footer>
+
+            Common errors to avoid:
+            1. The commit header <type>(<scope>): <subject> must be in lower case.
+            2. Allowed type values are: build, ci, docs, feat, fix, perf, refactor, test.
+
+            The linting rules are defined in:
+            https://github.com/calimero-network/core/blob/master/commitlint.config.js
+
+            More about the Commit Message Convention
+            https://github.com/calimero-network/core/blob/master/CONTRIBUTING.md#commit-message-style
+          '
+
+          echo "$PR_TITLE" | npx commitlint \
+            --config commitlint.config.js \
+            --help-url https://github.com/calimero-network/core/blob/master/CONTRIBUTING.md#commit-message-style || FAILED='true'
+
+          if [ -n "$FAILED" ]; then
+            jq -n \
+              --arg pr '${{ github.event.number }}' \
+              --arg tag commit-lint-check \
+              --arg mode "recreate" \
+              --arg message "$ERR" \
+              '{pr: $pr, tag: $tag, mode: $mode, message: $message}' | tee payload.json
+            exit 1
+          else
+            jq -n \
+              --arg pr '${{ github.event.number }}' \
+              --arg tag commit-lint-check \
+              --arg mode "delete" \
+              --arg message "" \
+              '{pr: $pr, tag: $tag, mode: $mode, message: $message}' | tee payload.json
+          fi
```

Comment:
can we deduplicate the jq call? like we do in [description check](https://github.com/calimero-network/core/blob/35c21784e3e785eee482c9f1cf5a59b66d1e2314/.github/workflows/pr-description-check.yml#L66-L71)